client/src/redux/actions/observations.js

```diff
@@ -1,57 +1,33 @@
-export const SAVING = 'SAVING';
-export const IDLE = 'IDLE';
-export const LOADING = 'LOADING';
-export const OBSERVATIONS_LOADED = 'OBSERVATIONS_LOADED';
-export const NUM_OBSERVATIONS = 'NUM_OBSERVATIONS';
-export const SET_OBSERVATIONS = 'SET_OBSERVATIONS';
-
-export function setNumObservations(numObservations) {
-  return {
-    type: NUM_OBSERVATIONS,
-    numObservations,
-  };
-}
-
-export function setObservations(observations) {
-  return {
-    type: SET_OBSERVATIONS,
-    observations,
-  };
-}
-
-export const addObservation = (observation, observationNumber) => (dispatch, getState) => {
-  const { observations } = getState().observationsReducer;
-  dispatch(setObservations(observations.concat(observation)));
-  dispatch(setNumObservations(observationNumber));
-};
-
-export const fetchObservations = userSession => async dispatch => {
-  const fileNames = [];
-  const observations = [];
-  let numObservations = 0;
-
-  // Get the observation file names
-  await userSession.listFiles(fileName => {
-    if (fileName.includes('observation/')) {
-      fileNames.push(fileName);
+import { getObservations, postObservation } from '../../utils/blockstackHelpers';
+import { clearSurvey } from './survey';
+
+export const ADD_OBSERVATION = 'ADD_OBSERVATION';
+export const FETCH_OBSERVATIONS = 'FETCH_OBSERVATIONS';
+
+export const addObservation = (userSession, observation) => async dispatch => {
+  const obs = await getObservations(userSession);
```

Comment:
So I'm just going to quickly outline what goes on here just to be sure everyone is on the same page...

Because our observations have to be stored as strings in gaia, we need to get all the observations from there, parse them into a JS array, add another item to that array, and stringify them again.

I think that what you're saying is in line with that, and it is just that you're saying rather than check the db for the current list of observations, we should check redux because we should already have the observations which we fetch when we first initialize the application.  I did consider this and the reason why I thought it was worth the extra call to gaia was the consideration of like... what if, for some unexpected and unlikely reason, our initial gaia data fetch had failed and we don't have that list in redux?  In that case, if we just check redux, and the new observation, and post that list to redux, we will have overwritten the observations file in gaia.

So I thought this approach was safety, and that was worth the extra call to gaia.

Thoughts?

Reply:
Makes sense. What do you think about checking Redux first, and if nothing is found, fetch from Gaia?

Reply:
I'm not sure that avoiding the one call to gaia seems with the added complexity?  But I'd be happy to hear others weigh in.

Reply:
I think it would be fine for now the way it is, but as we hopefully grow it may be worth adding the complexity now to cut down on network requests where we can to keep things snappy